compiler-rt/lib/scudo/standalone/vector.h

```diff
@@ -116,18 +118,19 @@ template <typename T> class VectorNoCtor {
   uptr CapacityBytes = 0;
   uptr Size = 0;
 
-  T LocalData[256 / sizeof(T)] = {};
+  T LocalData[StaticCapacity] = {};
   MemMapT ExternalBuffer;
 };
 
-template <typename T> class Vector : public VectorNoCtor<T> {
+template <typename T, size_t StaticCapacity = 8>
```

Comment:
Isn't this changing the default size of the vector?

I think it would be better to do something like:

StaticCapacityBytes = 256

Then above would be:

  T LocalData[StaticCapacityBytes / sizeof(T)] = {}

Reply:
Resolved by allowing user to configure vector size with bytes rather than the number of entries that can fit in local data.

Reply:
We have another buffer implementation in [release.h](https://github.com/llvm/llvm-project/blob/main/compiler-rt/lib/scudo/standalone/release.h#L102) and it gives the number of elements in the array. Also according to how we use std::array, std::vector or llvm::SmallVector, I would think having the number of elements is more close to what we usually do. 

I'm reading cferris suggesting as we may want to keep the default size. Instead of changing the parameter to number of bytes, I guess we want to change the default size instead. Which means, we can do it like

```
template <typename T, size_t StaticCapacity> class VectorNoCtor {
  // ...
T LocalData[StaticCapacity] = {};
};

template <typename T, size_t StaticCapacity = StaticCapacityBytes / sizeof(T)>
class Vector : public VectorNoCtor<T, StaticCapacity> {
  // ...
};
```
@cferris1000, what do you think?

Reply:
After thinking about this, I do think Chia-hung is right that we should use number of entries as the value to pass. Josh, sorry to keep having you make changes back and forth. I think it would be best to use number of entries everywhere though and ditch the Bytes version completely. However, I think we should remove the default and have everyone who use the vector be explicit about the sizes they use. This would require adding entry numbers where all vectors are used but I think that might be the best idea, rather than choosing some arbitrary default.

Also, instead of StaticCapacity as the name, maybe it would be better to use StaticNumEntries so it's clear what this value means.

Reply:
Agree with the removing of default and the new name

Reply:
I've now committed the changes. For the entry numbers for the vector instances currently in use, I simply set them to what they would have been under the previous initial capacity (i.e. 256 / sizeof(T) entries).